--- cmake/onnxruntime_unittests.cmake.orig
+++ cmake/onnxruntime_unittests.cmake
@@ -816,11 +816,11 @@ if (onnxruntime_USE_TVM)
   endif()
 endif()
 
-install(TARGETS onnx_test_runner
-        ARCHIVE  DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        LIBRARY  DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        BUNDLE   DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        RUNTIME  DESTINATION ${CMAKE_INSTALL_BINDIR})
+#install(TARGETS onnx_test_runner
+#        ARCHIVE  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+#        LIBRARY  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+#        BUNDLE   DESTINATION ${CMAKE_INSTALL_LIBDIR}
+#        RUNTIME  DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 if (NOT onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
   if(onnxruntime_BUILD_BENCHMARKS)
